File: src/main/java/com/reservapp/backend/model/Ciudad.java
package com.reservapp.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "ciudades")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Ciudad {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_provincia", nullable = false)
    private Provincia provincia;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(length = 10)
    private String codPostal;

    @OneToMany(mappedBy = "ciudad")
    private List<Empresa> empresas = new ArrayList<>();
}

File: reservapp_mobile/lib/models/ciudad.dart
class Ciudad {
  final int id;
  final String nombre;
  final String codPostal;
  final int idProvincia;

  const Ciudad({
    required this.id,
    required this.nombre,
    required this.codPostal,
    required this.idProvincia,
  });

  factory Ciudad.fromJson(Map<String, dynamic> json) => Ciudad(
    id: json['id'] ?? 0,
    nombre: json['nombre'] ?? '',
    codPostal: json['codPostal'] ?? '',
    idProvincia: json['idProvincia'] ?? 0,
  );
}

class Provincia {
  final int id;
  final String nombre;

  const Provincia({required this.id, required this.nombre});

  factory Provincia.fromJson(Map<String, dynamic> json) =>
      Provincia(id: json['id'] ?? 0, nombre: json['nombre'] ?? '');
}
